language/go/generate.go

```diff
@@ -712,7 +712,15 @@ func (g *generator) commonVisibility(importPath string) []string {
 		parent := strings.TrimSuffix(g.rel[:relIndex], "/")
 		visibility = append(visibility, fmt.Sprintf("//%s:__subpackages__", parent))
 	} else if importIndex >= 0 {
+		// This entire module is within an internal directory.
+		// Identify other repos which should have access too.
 		visibility = append(visibility, "//:__subpackages__")
+		for _, repo := range g.c.Repos {
+			if strings.HasPrefix(repo.AttrString("importpath"), importPath[:importIndex]) {
```

Comment:
Use `pathtools.HasPrefix` instead of `strings.HasPrefix`. It will automatically trim the trailing slash and guard against partial element matches.

Reply:
Good catch, done